### PR TITLE
Document softdevice and improve docs overall

### PR DESCRIPTION
**Pull request by TheButlah**
~Not yet ready for review.~

### CODE REVIEW

**Comment by TheButlah** [comment]
Still need to actually add softdevice support into the firmware. But this handles documentation of how the values were determined, and configures the memory.x and softdevice crates correctly.

File: firmware/docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+To make things easy, we have set things up to use [Cargo Embed].
+This provides flashing, log/RTT monitoring, and debugging all in one!
+
+Simply run:
+```
+cargo embed
+```
+This will flash the device and show you the log output. You can also do `cargo embed monitor`
+if you don't want to flash and merely want to view logs/debug.
+
```
Comment (Noxime) [inline firmware/docs/CONTRIBUTING.md:28]: There should be a bit that talks about the ESP32-C3 and using espflash

**Review by Noxime — APPROVED** [review]
Looks good to me.

File: firmware/docs/CONTRIBUTING.md (hunk 1 above)
Comment (TheButlah) [reply firmware/docs/CONTRIBUTING.md:28]: haven't added it yet

**Pull request merged**